cli-gui-integration.Rmd

```diff
@@ -1,8 +1,10 @@
 # CLI/GUI integration
 
-An important feature of R is it's command-line interface (CLI).
+An important feature of R is it's command-line interface (CLI). To be more specific, R has been designed from the beginning as an interface to other languages [@Chambers.2016]. By contrast, R has never been intended to be a GIS. This in mind, it is really astonishing how much geofunctionality R is providing today. Nevertheless, many, if not most geoalgorithms (despite rgdal, rgeos, sf, sp, etc.) are missing in R. Even worse, R's memory management almost inhibits the complex processing of big spatial data. Of course, it might be possible to overcome these shortcomings by writing new geo-related R packages which in turn are using C++ power in the background (cite Rcpp). However, that would only duplicate already available and already tested functionality, namely GIS functionality (most of which is written in C++ by the way). Hence, instead of duplicating, we merely need to create an interface to already existing software. This chapter will outline how you can access QGIS (**RQGIS**), SAGA (**RSAGA**) and GRASS (**rgrass7**) from within R, alongside good knowledge of the strengths and weaknesses of 'CLIs vs GUIs'.
```

Comment:
Interesting point that it was designed to be an interface language, to bring together C and FORTRAN as one person put it.

I think the additional content of this introductory paragraph is good and important but would suggest a few changes, to improve clarity, link to the software, use less hyperbole and more established terms (`geofunctionality` is not in my dictionary! and `really astonishing` sounds a little over-the-top). I would reword the paragraph as something like the following (note the use of one sentence per line while we're in this drafting stage of the book's development):

```
An important feature of R is it's command-line interface (CLI).
The language was originally designed as an interface to and extension of other languages, especially C and FORTRAN, to enable access to statistical algorithms in a user-friendly and intuitive read-evaluate-print loop (REPL)  [@Chambers.2016].
R was not intended to be a GIS, making the breadth of R's geospatial functionality astonishing to many who are unaware of its ability to replicate, and in some cases outperform, established GIS software for many operations on spatial data.

Instead of implementing existing GIS algorithms in R, it makes sense to avoid 'reinventing the wheel' by taking advantage of R's ability to interface with other languages (especially C++, which is used for much low-level and high-performance GIS work).
Using compiled code for new geoprocessing functionality (particularly with the help of the excellent **Rcpp** package) could form the basis of new R packages, building on the success of **sf**.
However, there is already a wide range of algorithms that can be accessed via R's interfaces to dedicated GIS software.
It makes sense to understand these before moving to develop your own optimised algorithms.
For this reason this chapter focusses on 'bridges' to the mature GIS products [QGIS](http://qgis.org/) (via the package **RQGIS**), [SAGA](http://www.saga-gis.org/) (**RSAGA**) and [GRASS](https://grass.osgeo.org/) (**rgrass7**) from within R.
By the end of the chapter you should have working knowledge of the functionality such packages open up, and a more nuanced understanding of the  'CLI vs GUI' debate.

```